odoo/addons/base/models/ir_default.py

```diff
@@ -117,7 +117,7 @@ def get(self, model_name, field_name, user_id=False, company_id=False, condition
         return json.loads(default.json_value) if default else None
 
     @api.model
-    @tools.ormcache('self.env.uid', 'model_name', 'condition')
+    @tools.ormcache('self.env.uid', 'self.with_context(prefetch_fields=False).env.company.id', 'model_name', 'condition')
```

Comment:
This one looks horrible and deserves the comment you added below.

Reply:
Comment moved as I removed the `prefetch_fields` below.